broadinstitute/rawls
# [WOR-407] Upgrade sentry and emit exceptions in global handler

## Context
Since we do not log most stack traces to our logs (due to quota management with logit, I believe) we are losing valuable exception information due to our sole reliance on Sentry logback. It would be helpful to have those stack traces for diagnostic purposes.



## This PR
* Adds `Sentry.captureException` calls to the Rawls Akka HTTP global exception handler. This will attach useful stack traces to exceptions that are emitted to Sentry.
* Upgrades sentry to 6.2.1. The `DefaultSentryClientFactory` class we were relying on for our custom event filtering is gone. The [guidance](https://docs.sentry.io/platforms/java/configuration/filtering/#filtering-error-events) from sentry is to pass in a callback during Sentry initialization, which I've done here.

## Testing 

I introduced a hardcoded `throw new RawlsExceptionWithErrorReport` to the `listWorkspaces` endpoint to test this. This is a sentry event without this change:
https://sentry.io/organizations/broad-institute/issues/3385818090/?project=116570&query=is%3Aunresolved

and after:
https://sentry.io/organizations/broad-institute/issues/3436054518/?project=116570&query=is%3Aunresolved



----------

  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
